src/asm/symbol.c

```diff
@@ -122,16 +122,50 @@ findsymbol(char *s, struct sSymbol * scope)
 {
 	struct sSymbol **ppsym;
 	SLONG hash;
-
-	hash = calchash(s);
+	int local;
+	int local2; // for testing the presence of another .
+	int result;
+
+	// If referencing a local label from out of scope,
+	// use the parent label as the new scope.
+	local = 0;
+	while (s[local] != '\0' && s[local] != '.') {
```

Comment:
Instead of doing this manually, it might be easier to use `strchr()`. Doing so would also eliminate the need for the `local2` variable.